### PR TITLE
docs: improved en version of tutorial_sql2

### PR DESCRIPTION
**Pull request by michelle-qinqin**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
fixed links
fixed one figure
added image source file
improved statement

### CODE REVIEW

File: docs/en/tutorial/tutorial_sql_2.md [hunk 1]
```diff
@@ -2,16 +2,20 @@
 
 ## 1. Preliminary Knowledge
 
-In the previous series of articles [Hands-On Tutorial for Feature Engineering Based on OpenMLDB (Part 1)](./tutorial_sql_1.md)）, we introduce the basic concepts and practical tools of Feature Engineering, as well as the basic feature script development based on single table. In this article, we will introduce in detail the more complex and powerful feature script development of the multi-table based on the main table and sub table. At the same time, we still rely on the SQL syntax provided by OpenMLDB for Feature Engineering script examples. For more information about OpenMLDB, please visit [GitHub repo of OpenMLDB](https://github.com/4paradigm/OpenMLDB), and the [document website](http://docs-cn.openmldb.ai/).
+In the previous series of articles [SQL for Feature Extraction (Part 1)](./tutorial_sql_1.md), we introduce the basic concepts and practical tools of feature engineering, as well as the basic feature processing script development based on single table. In this article, we will introduce more complex and more powerful feature processing script development on multi-tables.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: a single table

File: docs/en/tutorial/tutorial_sql_2.md [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
+In the previous series of articles [SQL for Feature Extraction (Part 1)](./tutorial_sql_1.md), we introduce the basic concepts and practical tools of feature engineering, as well as the basic feature processing script development based on single table. In this article, we will introduce more complex and more powerful feature processing script development on multi-tables. 
+At the same time, we still rely on the SQL syntax provided by OpenMLDB in feature engineering script examples. For more information about OpenMLDB, please visit [GitHub repo of OpenMLDB](https://github.com/4paradigm/OpenMLDB), and the [document website](https://openmldb.ai/docs/en/main/).
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: we still rely on the SQL syntax provided by OpenMLDB in feature engineering script examples
we will take the SQL syntax of OpenMLDB as an example for SQL-based feature engineering development.

File: docs/en/tutorial/tutorial_sql_2.md [hunk 3]
```diff
@@ -2,16 +2,20 @@
...
 
-If you want to run the SQL illustrated in this tutorial, please follow the following two steps to prepare:
+If you want to run the SQL in this tutorial, please follow the following two steps to prepare:
 
-- It is recommended to use the OpenMLDB docker image to run this tutorial under **Stand-alone Version**. Refer to [OpenMLDB Quick Start](http://docs-cn.openmldb.ai/2620852) for the operation mode. If using the clustered version, please use the offline mode（`SET @@execute_mode='offline'` ）. The common online mode of the cluster version only supports the simple data preview function, so most of the SQL in the tutorial cannot be run.
+- It is recommended to use the OpenMLDB docker image under **Stand-alone Version**. Refer to [OpenMLDB Quick Start](../quickstart/openmldb_quickstart.md) for the operation mode. If using the cluster version, please use the **Offline** mode (`SET @@execute_mode='offline'` ). The CLI tool of cluster version is only available under **Offline** mode and **Online Preview** mode. The Online Preview mode only can be used for simple data preview, so most of the SQL in the tutorial cannot run in this mode.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: **Stand-alone Version**
Standalone

File: docs/en/tutorial/tutorial_sql_2.md [hunk 4]
```diff
@@ -2,16 +2,20 @@
...
 - All data related to this tutorial and the import operation script can be downloaded [here](https://openmldb.ai/download/tutorial_sql/tutoral_sql_data.zip).
 
-In this article, we will use the main table and sub table to illustrate. We still use the sample data of anti-fraud transactions in the previous article, including a main table user transaction table (Table 1, t1) and a sub table merchant flow table (Table 2, t2). In order to avoid the consistency of multiple data tables in the database design, it is necessary to store multiple data tables in the database design according to the general design principle. In Feature Engineering, in order to obtain enough effective information, data needs to be extracted from multiple tables thus, Feature Engineering needs to be carried out based on multiple tables.
+In this article, we will use the main table and sub table to illustrate. We still use the sample data of anti-fraud transactions in the previous article, including a main table: user transaction table (Table 1, t1) and a sub table: merchant flow table (Table 2, t2).
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: we will use the main table and sub table to illustrate =>
the data set consists of a main table (holding the labels) and other secondary tables.
(change sub table to secondary table in this doc thoroughly)

File: docs/en/tutorial/tutorial_sql_2.md [hunk 5]
```diff
@@ -26,7 +30,7 @@ In this article, we will use the main table and sub table to illustrate. We stil
 | city       | STRING    | City                                    |
 | label      | BOOL      | Sample label, true\|false               |
 
-**Sub Table: Table 2, Merchant Flow Table t2**
+**Table 2: Merchant Flow Table, t2**
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: Merchant Transaction Table

File: docs/en/tutorial/tutorial_sql_2.md [hunk 6]
```diff
@@ -36,13 +40,13 @@ In this article, we will use the main table and sub table to illustrate. We stil
 | purchase_amt  | DOUBLE    | Purchase Amount                  |
 | purchase_type | STRING    | Purchase Type: Cash, Credit Card |
 
-In the traditional relational database, in order to obtain the information of multiple tables, the most common way is to use join for splicing. However, for the requirements of Feature Engineering, the database join can not meet the requirements very efficiently. The main reason is that our main table sample table has a label column for model training, and each value can only correspond to one row of data records. Therefore, in practice, we hope that after join, the number of rows in the result table should be consistent with that in the main table.
+In the traditional relational database, the most common way to obtain the information from multiple tables is to use JOIN. However, in feature engineering, JOIN of common database can not meet the efficiency requirement. The main reason is that our main sample table has a **label** column for training, and each value of this column only correspond to one row of the data records. We hope that the number of rows in the result table should be consistent with that in the main table after JOIN operation.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: i fee the above paragraph is not such accurate, we may remove this paragraph first.

File: docs/en/tutorial/tutorial_sql_2.md [hunk 7]
```diff
@@ -36,13 +40,13 @@ In this article, we will use the main table and sub table to illustrate. We stil
...
 
-## 2. Sub Table Single Line Feature
+## 2. Single Line Feature in Sub Table
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: change to "## 2. LAST JOIN"
remove the below "## 2.1 LAST JOIN"

(as only one section, no need a sub-title of 2.1)

File: docs/en/tutorial/tutorial_sql_2.md [hunk 8]
```diff
@@ -36,13 +40,13 @@ In this article, we will use the main table and sub table to illustrate. We stil
...
 
-## 2. Sub Table Single Line Feature
+## 2. Single Line Feature in Sub Table 
 
 ## 2.1 LAST JOIN
 
-OpenMLDB currently supports`LAST JOIN`to perform database like join operations. LAST JOIN can be regarded as a special LEFT JOIN. On the premise of meeting the JOIN conditions, each row of the left table will spell the last row that meets the conditions. LAST JOIN is divided into disordered splicing and ordered splicing. Let's take a simpler table as an example and assume that the schemas of tables s1 and s2 are
+OpenMLDB supports`LAST JOIN` to perform common JOIN operation in traditional database. LAST JOIN can be regarded as a special LEFT JOIN. Each row of the left table will concat the last row that meets the conditions of the right table. There are two types of LAST JOIN: disordered splicing and ordered splicing. Let's take a simple table as an example and assume that both the schemas of tables s1 and s2 are
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: OpenMLDB supports`LAST JOIN` to perform common JOIN operation in traditional database. LAST JOIN can be regarded as a special LEFT JOIN. 
=>
OpenMLDB supports `LAST JOIN` as a variant of LEFT JOIN that is supported in the standard SQL.

Comment (lumianph) [reply docs/en/tutorial/tutorial_sql_2.md]: Each row of the left table will concat the last row that meets the conditions of the right table.
=>
For each record in the left table, when there are multiple records matched from the right table, only one record will be output as the result.

Comment (lumianph) [reply docs/en/tutorial/tutorial_sql_2.md]: disordered splicing and ordered splicing =>
ordered and non-ordered LAST JOIN.

File: docs/en/tutorial/tutorial_sql_2.md [hunk 9]
```diff
@@ -51,98 +55,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
 Then, we can do the following join operation:
 
 ```sql
--- des c: In view of LAST JOIN splicing based on ORDER BY
 SELECT * FROM s1 LAST JOIN s2 ORDER BY s2.std_ts ON s1.col1 = s2.col1;
 ```
 
-As shown below, when `LAST JOIN` configure in `ORDER BY`，right click `std_ts` sort and splice the last hit data row. Take the second behavior of the left table as an example. There are 2 qualified right tables. Press `std_ts` after sorting and select the last item `3, b, 2020-05-20 10:11:13` .
+As shown below, left table `LAST JOIN` right table with `ORDER BY` and right table is sorted by  `std_ts`. Take the second row of the left table as an example. There are 2 qualified rows in right table. After sorting by `std_ts` the last row `3, b, 2020-05-20 10:11:13` is selected.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: There are 2 qualified
There are two matched

File: docs/en/tutorial/tutorial_sql_2.md [hunk 10]
```diff
@@ -51,98 +55,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
 
 ![img](images/lastjoin_1.jpg)
 
 ![img](images/lastjoin_2.jpg)
 
-## 3. Multi-Row Aggregation Feature of Sub Table
+## 3. Multi-Row Aggregation of Sub Table
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: ## 3. Multi-Row Aggregation over Multiple Tables

File: docs/en/tutorial/tutorial_sql_2.md [hunk 11]
```diff
@@ -51,98 +55,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
 
-For the sub table splicing scenario, OpenMLDB extends the standard window syntax and adds [WINDOW UNION]( http://docs-cn.openmldb.ai/2620896) characteristic. It supports splicing multiple pieces of data from the sub table to form a sub table window. Based on the sub table splicing window, it is convenient to construct the multi-row aggregation feature of the sub table. Similarly, two steps need to be completed to construct the multi-row aggregation feature of the sub table:
+For the sub table splicing scenario, OpenMLDB extends the standard WINDOW syntax and adds [WINDOW UNION](../reference/sql/dql/WINDOW_CLAUSE.md#window-union) syntax.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: For aggregation over multiple tables, ...

**Review by lumianph — CHANGES_REQUESTED** [review]
here is the first batch of comments, i will take a look at the remaining part later.

one more comment, do you have the source file of figures? the text on the figures is still in Chinese

**Comment by michelle-qinqin** [comment]
> here is the first batch of comments, i will take a look at the remaining part later.
> 
> one more comment, do you have the source file of figures? the text on the figures is still in Chinese

I will translate these figures tomorrow

File: docs/en/tutorial/tutorial_sql_2.md (hunk 6 above)
Comment (michelle-qinqin) [reply docs/en/tutorial/tutorial_sql_2.md]: I have removed it in en version, should I also removed the zh version?

File: docs/en/tutorial/tutorial_sql_2.md [hunk 12]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
+WINDOW UNION supports splicing multiple pieces of data from the secondary table to form a window on secondary table.
+Based on the secondary table splicing window, it is convenient to construct the multi-row aggregation feature of the secondary table. 
+Similarly, two steps need to be completed to construct the multi-row aggregation feature of the secondary table:
 
-- Step 1: Define the sub table splicing window.
-- Step 2: Construct the multi-row aggregation feature of the sub table on the sub table splicing window.
+- Step 1: Define the secondary table splicing window.
+- Step 2: Construct the multi-row aggregation feature of the secondary table on the window constructed in Step 1.
 
 ## 3.1 Step 1: Define the Sub Table Splicing Window
```
Comment (michelle-qinqin) [inline docs/en/tutorial/tutorial_sql_2.md]: should I change this h2 format to h3? line 169 likewise?
If the change is necessary, then I will change both the zh and en version

Comment (lumianph) [reply docs/en/tutorial/tutorial_sql_2.md]: yes, please fix it. thanks.

File: docs/en/tutorial/tutorial_sql_2.md [hunk 13]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
 
-The syntax of sub table splicing window is defined as:
+The syntax of creating a secondary table splicing window is defined as:
 
 ```sql
 window window_name as (UNION other_table PARTITION BY key_col ORDER BY order_col ROWS_RANGE｜ROWS BETWEEN StartFrameBound AND EndFrameBound)
 ```
 
-Among them, the most basic and indispensable grammatical elements include:
+Among them, the most basic and indispensable elements include:
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: most basic and indispensable => necessary

File: docs/en/tutorial/tutorial_sql_2.md [hunk 14]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
 
-- `UNION other_table`： `other_table` refers to the secondary table for WINDOW UNIOB. The primary and secondary tables need to keep the schema consistent. In most cases, the schema of the primary table and the secondary table are different. Therefore, we can ensure that the schema of the primary and secondary tables involved in window calculation is consistent by column filtering and default column configuration for the primary and secondary tables. Column filtering can also remove useless columns and only do WINDOW UNION and aggregation on key columns.
+- `UNION other_table`: `other_table` refers to the secondary table for WINDOW UNION. The primary and secondary tables' schemas need to keep consistent. In most cases, the schema of the primary table and the secondary table are different. We can ensure the schema consistency of the primary and secondary tables involved in window calculation by column filtering and default column configuration. Column filtering can also remove useless columns and only do WINDOW UNION and aggregation on important columns.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: column filtering
column projection

Comment (lumianph) [reply docs/en/tutorial/tutorial_sql_2.md]: two occurrences

File: docs/en/tutorial/tutorial_sql_2.md [hunk 15]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
 
-- `PARTITION BY key_col`: Indicates by column `key_col` splice matching data from the secondary table.
+- `PARTITION BY key_col`: Indicates that data is spliced by column `key_col` from the secondary table.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: maybe we change all splice to combine?

File: docs/en/tutorial/tutorial_sql_2.md [hunk 16]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
-- `ROWS_RANGE BETWEEN StartFrameBound AND EndFrameBound`: Represents the time interval of the sub table splicing window
+- `ROWS_RANGE BETWEEN StartFrameBound AND EndFrameBound`: Represents the time interval of the secondary table splicing window.
 
-- - `StartFrameBound`Represents the upper bound of the window.
+  - `StartFrameBound` represents the upper bound of the window.
 
-  - - `UNBOUNDED PRECEDING`: No upper bound.
-    - `time_expression PRECEDING`: If it is a time interval, you can define a time offset. For example, `30d preceding` means that the upper bound of the window is the time of the current line - 30 days.
+    - `UNBOUNDED PRECEDING`: There is no upper bound.
+    - `time_expression PRECEDING`: If it is a time interval, you can define a time offset. For example, `30d preceding` means that the upper bound of the window is 30 days before the time of the current line.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: current row?

File: docs/en/tutorial/tutorial_sql_2.md [hunk 17]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
-- `ROWS BETWEEN StartFrameBound AND EndFrameBound`: Represents the time interval of the sub table splicing window.
+- `ROWS BETWEEN StartFrameBound AND EndFrameBound`: Represents the time interval of the secondary table splicing window.
 
-- - `StartFrameBound`Represents the upper bound of the window.
+  - `StartFrameBound` represents the upper bound of the window.
 
-  - - `UNBOUNDED PRECEDING`: No upper bound.
-    - `number PRECEDING`: If it is a number interval, you can define the number of time items. For example, `100 PRECEDING` indicates the first 100 lines of the current line whose upper bound is.
+    - `UNBOUNDED PRECEDING`: There is no upper bound.
+    - `number PRECEDING`: If it is a number interval, you can define the number of rows. For example, `100 PRECEDING` indicates that the upper bound is 100 lines before the current line.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md]: current row

File: docs/en/tutorial/tutorial_sql_2.md [hunk 18]
```diff
@@ -51,98 +52,111 @@ OpenMLDB currently supports`LAST JOIN`to perform database like join operations.
...
+ 
+  - `EndFrameBound` Represents the lower bound of the window.
 
-  - `EndFrameBound`Represents the lower bound of the time window.
+    - `CURRENT ROW`: The lower bound is current row.
+    - `number PRECEDING`: If it is a number interval, you can define the number of rows. For example, `1 PRECEDING`  indicates that the lower bound is 1 line before the current line.
```
Comment (lumianph) [inline docs/en/tutorial/tutorial_sql_2.md:114]: current row

**Review by lumianph — CHANGES_REQUESTED** [review]

**Review by lumianph — APPROVED** [review]

**Pull request merged**